PathManager/Inc/PM_States.hpp

```diff
@@ -0,0 +1,226 @@
+        commsWithTelemetry& operator =(const commsWithTelemetry& other);
+        static Telemetry_PIGO_t incomingData; // Stores the commands sent by telemetry for easy access by other states in the pathmanager
+};
+
+class SensorFusion : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static SFOutput_t* GetSFOutput(void) { return &sfOutputData; }
+        static IMU_Data_t* GetIMUData(void) { return &imudata; }
+    private:
+        sensorFusion() {}
+        sensorFusion(const sensorFusion& other);
+        sensorFusion& operator =(const sensorFusion& other);
+        static SFOutput_t sfOutputData;
+        static IMU_Data_t imudata;
+        // If additional raw sensor data is needed, add the static struct here and implement
+        // it like the IMU_Data_t struct above
+};
+
+*/
+
+
+class CommsWithAttitude : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static CommandsForAM* GetCommFromAttitude(void) { return &receivedData; }
+    private:
+        commsWithAttitude() { CommFromPMToAMInit(); } // Initializes module
+        commsWithAttitude(const commsWithAttitude& other);
+        commsWithAttitude& operator =(const commsWithAttitude& other);
+        static CommandsForAM receivedData;
+};
+
+class CommsWithSystemManager : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static CommandsFromSM* GetSMIncomingData(void) {return &_incomingData;}
+    private:
+        CommsWithSystemManager() {CommWithSMInit();}
+        CommsWithSystemManager(const CommsWithSystemManager& other);
+        CommsWithSystemManager& operator =(const CommsWithSystemManager& other);
+        static CommandsFromSM incomingData; // Stores the commands sent by telemetry for easy access by other states in the pathmanager
+};
+
+
+class FlightModeSelector : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        flightModeSelector() {}
+        flightModeSelector(const flightModeSelector& other);
+        flightModeSelector& operator =(const flightModeSelector& other);
+
+};
+
+/* TODO: Talk w Gordon about how we should structure takeoff/landing / what is needed */
+
+class PreflightStage : public pathManagerState
```

Comment:
This can be removed. If we need preflight then SM can handle it pretty easily and that makes more sense as it won't need any waypoints. Also that would be after MVP.